src/preamble.js

```diff
@@ -1034,7 +1034,7 @@ function createWasm() {
     exportWasmSymbols(wasmExports);
 #endif
 
-#if PTHREADS || WASM_WORKERS
+#if SHARED_MEMORY
```

Comment:
This change, and the change above are not strictly correct under the meaning of `SHARED_MEMORY`, but it should not be an issue either (will just increase code size for users of `-sSHARED_MEMORY=1`)

There are three different ways to target multithreading:

- `-sPTHREADS=1`, which implies `SHARED_MEMORY=1`
- `-sWASM_WORKERS=1`, which implies `SHARED_MEMORY=1`
- `-sSHARED_MEMORY=1`, which is used in situations where Emscripten compiled build should run single-threaded but using a shared WebAssembly Memory, so that developers can develop custom (non-Emscripten-compiled) Workers

In other words, `#if SHARED_MEMORY` does not mean "using pthreads or using wasm workers", but it means "using pthreads, wasm workers or manual shared memory".

The original `#if PTHREADS || WASM_WORKERS` would be more accurate here. Likewise above, the function `resetPrototype` would not be used in `-sSHARED_MEMORY=1` builds.

Reply:
Revert to using `#if PTHREADS || WASM_WORKERS`.